### DISCUSSION TITLE
2 story windows?

### DISCUSSION THREAD

**Discussion by cgray-bt** [basbruss/adaptive-cover#336] [Q&A]
I have this all set up and it's doing what i think it should.  Amazing BTW, thank you!

But my windows are a double stack in a 2 story living room.  So the windows with shades are the upper bank, 6ft tall windows where the bottom of the windows are about 8-9 feet off the ground.  So it seems like it should be closing the vertical shades more than it is because the sunlight hits so far away and angles down much further than if the windows were at eye level.  

Is there any config that could adjust for that?  Or am I thinking about it incorrectly?

**Comment by sh-nguyen** [comment]
From the code it looks like the window's vertical position is not taken into account when calculating. Perhaps we could change the logic from this:
`blind_height = np.clip(
            (self.distance / cos(rad(self.gamma))) * tan(rad(self.sol_elev)),
            0,
            self.h_win,
        )`
to this:
`blind_height = np.clip(
            ((self.distance / cos(rad(self.gamma))) * tan(rad(self.sol_elev))) - self.p_win,
            0,
            self.h_win,
        )`
where self.p_win is the vertical position of the bottom of the window with respect to the work area.
